lib/system.nim

```diff
@@ -3124,5 +3124,12 @@ when not defined(createNimHcr) and not defined(nimscript):
 when notJSnotNims and not defined(nimSeqsV2):
   proc prepareStrMutation*(s: var string) {.inline.} =
     ## String literals (e.g. "abc", etc) in the ARC/ORC mode are "copy on write",
-    ## therefore you should call `prepareStrMutation` before modifying the strings.
+    ## therefore you should call `prepareStrMutation` before modifying the strings
+    ## via `addr`.
+    runnableExamples("--gc:arc"):
+      var x = "abc"
+      var y = x
+      prepareStrMutation(y)
+      moveMem(addr y[0], addr x[0], 3)
+      assert y == "abc"
```

Comment:
better example:
```suggestion
      var x = "abc"
      var y = "defgh"
      prepareStrMutation(y)
      moveMem(addr y[0], addr x[0], x.len)
      assert y == "abcgh"
```
(example you had worked even without calling `prepareStrMutation(y) and moveMem(addr y[0], addr x[0], 3)`; and new example is less special cased)

Reply:
Really, do you compile with `--gc:arc`?

Reply:
the original example you gave wasn't good because it worked without `prepareStrMutation`: this works (even with --gc:arc):
```nim
when true:
      var x = "abc"
      var y = x
      moveMem(addr y[0], addr x[0], 3)
      assert y == "abc"
```
the example i gave works only with `prepareStrMutation` (with --gc:arc), otherwise gives SIGBUS

Reply:
EDIT: looks like this is OS-specific (@xflywind reports SIGSEGV instead of no error on windows); we should track this, I'd prefer deterministic crash on all platforms (maybe)

Reply:
![image](https://user-images.githubusercontent.com/43030857/109614337-7769b480-7b6d-11eb-8710-043b74e80b4b.png)